index.bs

```diff
@@ -551,6 +559,59 @@ It must act as follows:
   </pre>
 </div>
 
+<h4 id="stream-method-algo">
+The stream method</h4>
+
+The <dfn method for=Blob>stream()</dfn> method returns the contents
+of the blob as a {{ReadableStream}}.
+
+It must return the result of [=getting a stream for a Blob=] for the [=context object=].
+
+To <dfn lt="get a stream for a Blob|getting a stream for a Blob">get a stream for a {{Blob}}</dfn>
+|blob|, run the following steps:
+
+1. Let |stream| be the result of [=construct a ReadableStream object|constructing=] a
+   {{ReadableStream}} object.
+1. Return |stream| and run the following steps [=in parallel=]:
+  1. While not all bytes of |blob| have been read:
+    1. Let |bytes| be the byte sequence that results from reading a [=chunk=] from |blob|.
+    1. If a [=file read error=] occured while reading |bytes|, [$ReadableStream/error$]
+       |stream| with a [=failure reason=] and abort these steps.
+    1. [=ReadableStream/Enqueue=] |bytes| into |stream|.
+
+    Issue: We need to specify more concretely what reading from a Blob actually does, and what
+    possible errors can happen.
+
+<h4 id="text-method-algo">
+The text method</h4>
+
+The <dfn method for=Blob>text()</dfn> method returns the contents
+of the blob as a {{USVString}}.
+
+It must act as follows:
+
+1. Let |stream| be the the result of [=getting a stream for a Blob=] for the [=context object=].
```

Comment:
So if we're going to define these in terms of streaming. Should we perhaps test this approach on `FileReader` first? So that the underlying streaming primitive is powerful enough to explain all these various features. For this particular method we could then likely use a shared abstraction as `FileReader` also has a string-returning feature.

Reply:
Sure, I don't particularly mind doing things in either order. This seemed like a smaller change to start with, but the opposite is of course true as well. I'm working on the more general FileReader rewrite, and should have that uploaded soon too.

This also brings up the question if `blob.text()` should do what is described here, matching `new Response(blob).text()` (and always using UTF-8), or if it should do what `readAsText` does currently, and use the encoding that might be specified in the blob's mime type (or even have an optional encoding parameter to pass to the text method to override that default). Not sure which one is more important to be consistent with...

Reply:
> This also brings up the question if blob.text() should

Oooh, interesting question. I tend to think it'd be a bit frustrating for authors and users if the charset was ignored, especially if interoperating with legacy systems or files on disk that are not properly UTF-8 encoded. But, I'm not sure.

Reply:
Any idea why `Body.text()` ignores the charset? It seems like it could be equally frustrating there, and the algorithm there does have access to the mime type as well.

Reply:
I did it because it's a convenience method and everyone should migrate to UTF-8.

It's not clear to me how useful it would be to consider other encodings here if you only look at the MIME type as file systems wouldn't give you that information at that level, presumably.

Reply:
https://github.com/w3c/FileAPI/pull/118 is my attempt at rewriting FileReader in similar terms as this pull request does. Let me know if you want me to finish landing that before getting back to this PR?

Just sticking to UTF-8 seems fine to me. At least it would be consistent with fetch.

Reply:
I'd prefer that, as it's the right way to approach these kind of problems. Ensure the new primitive can be used to implement all the things, then expose fancy new APIs.